src/FakeItEasy/ArgumentEqualityComparer.cs

```diff
@@ -0,0 +1,31 @@
+namespace FakeItEasy;
+
+using System;
+
+public abstract class ArgumentEqualityComparer<T> : IArgumentEqualityComparer
+{
+    /// <summary>
+    /// Gets the priority of the argument equality comparer. When multiple comparer that apply to the same type are registered,
+    /// the one with the highest priority value is used.
+    /// </summary>
+    public virtual Priority Priority => Priority.Default;
+
+    /// <summary>
+    /// Whether or not this object can compare instances of <paramref name="type"/> for equality.
+    /// </summary>
+    /// <param name="type">The type of the objects to compare.</param>
+    /// <returns>
+    /// <c>true</c> if this object can compare instances of <paramref name="type"/>. Otherwise <c>false</c>.
+    /// </returns>
+    public virtual bool CanCompare(Type type) => type == typeof(T);
+
+    /// <summary>
+    /// Gets a value indicating if <paramref name="expectedValue"/> and <paramref name="argumentValue"/> are considered equal.

```

Comment:
Nothing wrong with this, but feels wordy. "Indicates whether…"?

Reply:
Oh. Is "Gets…" mandated by some style rule?

Reply:
I don't think it's mandated by any style rule. I'll see what I can do.